statsrelay/src/backend_client/prom_client.rs

```diff
@@ -419,4 +434,30 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn prom_client_abort_retries() {
+        let config: PromBackendClientConfig = PromBackendClientConfig {
+            max_queue: 10000,
+            request_timeout_ms: 10000,
+            batch_creation_timeout_ms: 500,
+            max_batch_size: 5,
+            max_in_flight: 1,
+        };
+        let mock_client = MockPromWriteRequestClient::new(hyper::StatusCode::INTERNAL_SERVER_ERROR);
+        let scope = Collector::default().scope("stats");
+        let endpoint = "endpoint";
+
+        let prom_client =
+            PromClient::new_with_client(scope, mock_client.clone(), endpoint, config.clone());
+
+        let pdu = Pdu::parse(Bytes::from_static(b"foo.bar:3|c|@1.0|#tags")).unwrap();
+
+        prom_client.try_send(pdu).expect("failed to send pdu");
+
+        let join_handle = prom_client.join_handle().unwrap();
+        drop(prom_client);
+        // This will block forever if we don't abort retries. 
+        let _ = join_handle.await;
```

Comment:
What's the behavior of this failing in CI -- is it worth wrapping this in `tokio::time::timeout(...)` so to turn the infinite block into failure after X seconds so we don't e.g. wait for the github actions timeout to kill the run?

Reply:
Ah thanks for catching that. I added a timeout.